packages/block/src/from-beacon-payload.ts

```diff
@@ -0,0 +1,65 @@
+import { bigIntToHex } from '@ethereumjs/util'
+
+import type { ExecutionPayload } from './types'
+
+type BeaconWithdrawal = {
+  index: string
+  validator_index: string
+  address: string
+  amount: string
+}
+
+// Payload json that one gets using the beacon apis
+// curl localhost:5052/eth/v2/beacon/blocks/56610 | jq .data.message.body.execution_payload
+export type BeaconPayload = {
+  parent_hash: string
+  fee_recipient: string
+  state_root: string
+  receipts_root: string
+  logs_bloom: string
+  prev_randao: string
+  block_number: string
+  gas_limit: string
+  gas_used: string
+  timestamp: string
+  extra_data: string
+  base_fee_per_gas: string
+  block_hash: string
+  transactions: string[]
+  withdrawals?: BeaconWithdrawal[]
+  excess_data_gas?: string
+}
+
+export function beaconToExecutionPayload(payload: BeaconPayload): ExecutionPayload {
```

Comment:
Can we rename this `beaconBlockToExecutionPayload` and add some docs to explain what it does and how to use it?

Reply:
its not a beacon block, its execution payload in beacon block ( see the jq path here: `curl localhost:5052/eth/v2/beacon/blocks/56610 | jq .data.message.body.execution_payload`), let me know what you finally want to keep

Reply:
So maybe name it something like `executionPayloadFromBeaconBlock` and then add TypeDocs specifying what the input/outputs are.  The doc could be something like:
```
/**
 * Converts a beacon block execution payload JSON object to the {@link ExecutionPayload} data needed to construct a {@link Block}.
 * The JSON data can be retrieved from a consensus layer (CL) client on this Beacon API `/eth/v2/beacon/blocks/[block number]`
 */
 ```

Reply:
applied changes